Assets/Scripts/Algorithms/Patrolling/CognitiveCoordinated/CognitiveCoordinatedBase.cs

```diff
@@ -27,34 +26,21 @@ protected override void GetDebugInfo(StringBuilder stringBuilder)
         }
 
         // Set by CreateComponents
-        private GoToNextVertexComponent _goToNextVertexComponent = null!;
-        private CollisionRecoveryComponent _collisionRecoveryComponent = null!;
-        private IRobotController _controller = null!;
+        protected GoToNextVertexComponent _goToNextVertexComponent = null!;
+        protected CollisionRecoveryComponent _collisionRecoveryComponent = null!;
+        protected IRobotController _controller = null!;
 
         protected override IComponent[] CreateComponents(IRobotController controller, PatrollingMap patrollingMap)
         {
             _controller = controller;
-            _goToNextVertexComponent = new GoToNextVertexComponent(NextVertex, this, controller, Coordinator.GlobalMap);
+            _goToNextVertexComponent = new GoToNextVertexComponent(NextVertex, this, controller, patrollingMap);
             _collisionRecoveryComponent = new CollisionRecoveryComponent(controller, _goToNextVertexComponent);
-
             return new IComponent[] { _goToNextVertexComponent, _collisionRecoveryComponent };
         }
 
-        public override void SetGlobalPatrollingMap(PatrollingMap globalMap)
-        {
-            base.SetGlobalPatrollingMap(globalMap);
-
-            // Will be set multiple times.
-            // Too bad!
-            Coordinator.GlobalMap = globalMap;
-
-            // We must clear Coordinators knowledge of occupied vertices as we have new robots, and we might just have started a new experiment.
-            // This will break if robots are added mid-experiment.
-            Coordinator.ClearOccupiedVertices();
-        }
-
-        private Vertex NextVertex(Vertex currentVertex)
+        protected Vertex NextVertex(Vertex currentVertex)
         {
+            UpdateLastTimeVisitedTick(currentVertex);
```

Comment:
Are you sure the robot has visited the vertex before this is called?

Reply:
I dont think it does

Reply:
You are right, but I've fixed it now :)

Reply:
Now I pass a stigmergy.put to the:
`public void OnReachTargetVertex(Vertex vertex, Vertex nextVertex)
        {
            TargetVertex = nextVertex;
            OnReachVertexHandler?.Invoke(vertex.Id);

            if (!AllowForeignVertices || (AllowForeignVertices && !_globalMap.Vertices.Contains(vertex)))
            {
                vertex.VisitedAtTick(_logicTicks);
            }
        }`